devel/namespaces/index.md

```diff
@@ -0,0 +1,169 @@
+
+* `worker-type:<provisionerId>/<workerType>` -
+   Roles of this form represent scopes available to workers of the given type.
+
+## Artifacts
+
+Artifacts are named objects attached to tasks, and available from the queue component.
+Artifact names are, by convention, slash-separated.
+
+* `public/` -
+   The queue allows access to any artifact that begins with `public/` without any kind of authentication.
+   Public names are not further namespaced: tasks can create any public artifacts they like.
+   As such, users should not assume that an artifact with this prefix was created by a known process.
+   In other words, any task can create an artifact named `public/build/firefox.exe` , so do not trust such a file without verifying the trustworthiness of the task.
+
+* `private/` -
+   Artifact names with this prefix are considered non-public, but access to them is otherwise quite broadly allowed (e.g., to all Mozilla employees).
+   In general, users with narrower requirements than "not public" should select a different prefix and add it to this document.
+
+* `index/` -
+   Artifact names with this prefix follow the index namespace, replacing the `.` separators with `/`.
+   For example, a task that indexes artifacts under `gaia.npm_cache.node-v4-2.linux-x64` would name its artifacts with the prefix `index/gaia/npm_cache/node-v4_2/linux-x64`.
+   Scopes for creating and reading such artifacts should follow the ownership hierarchy of the index routes.
+
+* `project/<project>/` -
+   Artifact names with this prefix are the responsibility of the project, which may have further namespace conventions.
+
+## Hooks
+
+Hooks are divided into "hook groups", for which the namespace is defined here.
+Within a hook group, the names are arbitrary (or defined by the project).
+
+* `project:<project>` - hooks for a specific project
+
+## Worker Types
+
+Worker types are broken down into `<provisionerId>` and `<workerType>`.
+Within the AWS provisioner, worker types are currently not namespaced.
+All names are up for grabs -- [for now](https://bugzilla.mozilla.org/show_bug.cgi?id=1220686).
+
+## Worker IDs
+
+Worker IDs are broken down into `<workerGroup>` and `<workerId>`.
+In the present implementation, both of these are arbitrary strings.
+For workers started by the AWS provisioner, they are avaiability zone and instance ID, respectively.
+For other worker types, anything goes.
+
+## Caches
+
+Caches are not namespaced, but they [should be](https://bugzilla.mozilla.org/show_bug.cgi?id=1220684).
+
+## Secrets
+
+Secrets provide key-value storage governed by scopes.
+As such, it's very important that secrets not be unexpectedly made accessible to users who should not see them.
+Secret names have the following structure:
+
+* `project/<project>/` -
+  Secrets with this prefix are the exclusive domain of the given project.
+  Users not associated with a project should not be given scopes associated with the project's secrets!
```

Comment:
I propose that we add a `repo:<...>` similar to the `repo:` role pattern you outlined earlier...